plugins/reports/counter/CounterReportPlugin.inc.php

```diff
@@ -59,6 +77,101 @@ function getTemplatePath() {
 		return parent::getTemplatePath() . 'templates/';
 	}	 
 
+
+	/**
+	 * List the valid COUNTER releases
+	 * Must exist in the release path as CounterRelease{version}.inc.php. If version is M_m, this will become M.m
+	 * @return array
+	 */
+	function getValidReleases() {
+		$releases = array();
+		$prefix = $this->getReleasePath().DIRECTORY_SEPARATOR.'CounterRelease';
+		$suffix = COUNTER_CLASS_SUFFIX;
+		foreach (glob($prefix.'*'.$suffix) as $file) {
+			if ($file == $prefix.$suffix) {
+				continue;
+			}
+			$releases[] = str_replace('_', '.', substr($file, strlen($prefix), -strlen($suffix)));
+		}
+		return $releases;
+	}
+
+	/**
+	 * Get the latest counter release
+	 * Must exist in the release path as CounterRelease{version}.inc.php
+	 * @return string
+	 */
+	function getLatestRelease() {
+		$releases = $this->getValidReleases();
+		sort($releases);
+		return array_pop($releases);
+	}
+	
+	/**
+	 * List the valid reports
+	 * Must exist in the report path as {Report}_r{release}.inc.php
+	 * @return array multidimentional array release => array( report => reportClassName )
+	 */
+	function getValidReports() {
+		$reports = array();
+		$prefix = $this->getReportPath().DIRECTORY_SEPARATOR;
+		foreach ($this->getValidReleases() as $release) {
+			$suffix = '_r'.str_replace('.', '_', $release).COUNTER_CLASS_SUFFIX;
+			foreach (glob($prefix.'*'.$suffix) as $file) {
+				$reports[$release][substr($file, strlen($prefix), -strlen($suffix))] = substr($file, strlen($prefix), -strlen(COUNTER_CLASS_SUFFIX));
```

Comment:
it's minor, but I think that naming (assigning to variables with meaningful names) the result of those substrings operations would help a lot to read the code.